data-prepper-plugins/armeria-common/src/test/java/com/amazon/dataprepper/plugins/GrpcAuthenticationProviderTest.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins;
+
+import com.amazon.dataprepper.armeria.authentication.GrpcAuthenticationProvider;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class GrpcAuthenticationProviderTest {
+
+    @Mock
+    private GrpcAuthenticationProvider provider;
+
+    @Test
+    public void confirmDefaultBehavior() {
+        when(provider.getHttpAuthenticationService()).thenCallRealMethod();
+
+        Assertions.assertNotNull(provider.getHttpAuthenticationService());
```

Comment:
This should be extracted into a variable and re-used. Otherwise you are "testing" twice.

```
Optional<Function<? super HttpService, ? extends HttpService>> actualHttpService = provider.getHttpAuthenticationService());
```

Personally I'm also fine using `var` here. But, I know that can be contentious.

```
var actualHttpService = provider.getHttpAuthenticationService());
```

Reply:
I'm all for `var`!